testProject/ViewController.m

```diff
@@ -0,0 +1,216 @@
+        }];
+    }
+}
+
+- (void)authenticateWithCompletion:(void (^)())completionBlock
+{
+    //Authenticate account
+    
+    NSURLSessionConfiguration *configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
+    AFHTTPSessionManager *manager = [[AFHTTPSessionManager alloc] initWithSessionConfiguration:configuration];
+    manager.requestSerializer = [AFJSONRequestSerializer serializer];
+    
+    NSData *plainData = [[NSString stringWithFormat:@"%@:%@", @"CFMvDaHhOCwBeqMlPKDjeBODG", @"rV3L8VRTgCepmbVX25zXsSYPNzfczSv6kyky6AhwuLkr2ETYw1"] dataUsingEncoding:NSUTF8StringEncoding];
+    NSString *encodedUsernameAndPassword = [plainData base64EncodedStringWithOptions:0];
+    NSString *authHeader = [NSString stringWithFormat:@"Basic %@", encodedUsernameAndPassword];
+    [manager.requestSerializer setValue:authHeader forHTTPHeaderField:@"Authorization"];
+    
+    NSString *URL = @"https://api.twitter.com/oauth2/token?grant_type=client_credentials";
+    [manager POST:URL parameters:nil progress:nil
+          success:^(NSURLSessionDataTask *operation, id responseObject) {
+              NSLog(@"Res: %@", responseObject);
+              self.tokenType = [responseObject valueForKeyPath:@"token_type"];
+              self.accessToken = [responseObject valueForKeyPath:@"access_token"];
+              
+              if (completionBlock) {
+                  completionBlock();
+              }
+          }
+     
+          failure:^(NSURLSessionDataTask *operation, NSError *error) {
+              NSLog(@"Auth error: %@", error);
+          }];
+}
+
+- (void)fetchAndDisplayTweets {
+    AFHTTPSessionManager *afSessionManager = [AFHTTPSessionManager manager];
+    NSString *authHead = [NSString stringWithFormat:@"%@ %@", self.tokenType, self.accessToken];
+    [afSessionManager.requestSerializer setValue:authHead forHTTPHeaderField:@"Authorization"];
+    
+    [afSessionManager GET:[self formattedUrlStringFromSearchField:self.isSearchPaged] parameters:nil progress:nil success:^(NSURLSessionTask *task, id responseObject) {
+        self.isSearchPaged = false;
+        if(responseObject)
+        {
+            
+            for(NSDictionary *dictStat in [responseObject objectForKey:@"statuses"]) {
+                
+                [self.users addObject:[dictStat valueForKeyPath:@"user.screen_name"]];
+                [self.tweets addObject:[dictStat valueForKeyPath:@"text"]];
+                [self.profilePictureUrls addObject:[dictStat valueForKeyPath:@"user.profile_image_url_https"]];
+                [self.tweetsTableView reloadData];
+            }
+            self.nextPageUrl = [responseObject valueForKeyPath:@"search_metadata.next_results"];
+            _pageAvailable = true;
+            
+        } else{
+            NSLog(@"Error did not receive response");
+        }
+        
+        
+    } failure:^(NSURLSessionTask *operation, NSError *error) {
+        NSLog(@"Error: %@", error);
+    }];
+}
```

Comment:
Ideally you want to have a manager class that will make the network calls for you. However, this is ok for the project.